packages/core/src/routes/application.test.ts

```diff
@@ -146,6 +151,33 @@ describe('application route', () => {
     ).resolves.toHaveProperty('status', 400);
   });
 
+  it('PATHC /applications/:applicationId should save the formatted URIs as per RFC', async () => {
+    await expect(
+      applicationRequest.patch('/applications/doo').send({
+        oidcClientMetadata: {
+          ...customClientMetadata,
+          redirectUris: [
+            'com.google.com://callback',
+            'https://example.com/callback?auth=true',
+            'http://example.com',
+            'http://localhost:3002',
+          ],
+        },
+      })
+    ).resolves.toHaveProperty('status', 200);
+  });
+
+  it('PATCH /application/:applicationId expect to thorw with invalid redirectURI', async () => {
+    await expect(
+      applicationRequest.patch('/applications/doo').send({
+        oidcClientMetadata: {
+          ...customOidcClientMetadata,
+          redirectUris: ['com.google.com'],
+        },
+      })
+    ).resolves.toHaveProperty('status', 400);
+  });
+
```

Comment:
We should add logic and test cases that only when app type is `Native`, the redirect URI is expected to contain a reverse domain as package name.
For SPA and traditional web applications, URIs start with `http` and `https` can be accepted, and no `.` needs to be included in the URI.